build.gradle.kts

```
@@ -47,4 +60,83 @@ val integrationTestImplementation: Configuration by configurations.getting {
 }
 val integrationTestRuntimeOnly: Configuration by configurations.getting {
     extendsFrom(configurations.testRuntimeOnly.get())
-}
\ No newline at end of file
+}
+val containerName = findProperty("containerName") ?: "hivemq-k8s-test"
+val containerTag = findProperty("containerTag") ?: "snapshot"
+
+val producerK8sDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("k8s-docker-image"))
+    }
+    extendsFrom(hivemq)
+}
+val producerDnsInitWaitDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("dns-init-wait-docker-image"))
+    }
+    extendsFrom(operator)
+}
+val producerOperatorDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("operator-docker-image"))
+    }
+    extendsFrom(operator)
+}
+
+
+val createRootlessK8sImageContext by tasks.registering(Sync::class) {
+    group = "container"
+    description = "Prepare hivemq rootless k8s image context"
+    into(layout.buildDirectory.dir("container/context"))
+    from("container")
+}
+
+val buildRootlessK8sImage by tasks.registering(Exec::class) {
+    group = "container"
+    description = "Build hivemq rootless k8s image"
+    inputs.property("dockerImageName", containerName)
+    inputs.dir(createRootlessK8sImageContext.map { it.destinationDir })
+    workingDir(createRootlessK8sImageContext.map { it.destinationDir })
+    commandLine("docker", "build", "-f", "broker.dockerfile", "-t", "${containerName}-rootless:${containerTag}", ".")
+}
```

```suggestion
    dependsOn(producerK8sDockerImage)
}
```
if the rootless image depends on the "non-rootless" image then please add an explicit dependency